lib/modules/apostrophe-login/index.js

```diff
@@ -305,12 +306,54 @@ module.exports = {
       return self.apos.users.safe.findOne({
         _id: user._id
       }).then(function(userSafe) {
-        return done(null, userSafe && userSafe.totp && userSafe.totp.key, userSafe && userSafe.totp && userSafe.totp.period);
+        return self.checkIfActive(userSafe, done);
```

Comment:
Code for TOTP (Google Authenticator) was removed here and did not come back anywhere else.

You need to pass an inline function as the callback of checkIfActive, and in that callback, do what the original code did.

Do not move that code into checkIfActive, as that would be poor separation of concerns. Just make sure the callback given to checkIfActive does the original job fully.

Reply:
Google Authenticator support should be retested with this code.

Reply:
Ok, tested with Google auth and working